### ISSUE TITLE
 Error: Unable to resolve module

### ISSUE THREAD

**Issue by Splicer97** [callstack/react-native-builder-bob#324]
### Description

When I was creating my library, yarn example (ios or android) works fine. But when I publish my package, delete them locale and download from github again, I get this error when trying run example (ios or android)

`
 Error: Unable to resolve module react-native-mytracker from /Users/user/Projects/react-native-mytracker/example/src/App.tsx: react-native-mytracker could not be found within the project or in these directories:
  node_modules
  ../node_modules
  ../../../node_modules
 import { StyleSheet, Text, View } from 'react-native';
> 4 | import MyTracker from 'react-native-mytracker';
    |                        ^
  5 |
  6 | export default function App() {
  7 |   MyTracker.initTracker('initTracker');
    at ModuleResolver.resolveDependency (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/node-haste/DependencyGraph/ModuleResolution.js:178:15)
    at DependencyGraph.resolveDependency (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/node-haste/DependencyGraph.js:264:43)
    at Object.resolve (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/lib/transformHelpers.js:170:21)
    at resolveDependencies (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/graphOperations.js:466:33)
    at processModule (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/graphOperations.js:232:31)
    at async traverseDependenciesForSingleFile (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/graphOperations.js:221:3)
    at async traverseDependencies (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/graphOperations.js:147:7)
    at async DeltaCalculator._getChangedDependencies (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/DeltaCalculator.js:263:42)
    at async DeltaCalculator.getDelta (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler/DeltaCalculator.js:90:16)
    at async DeltaBundler.getDelta (/Users/user/Projects/react-native-mytracker/example/node_modules/metro/src/DeltaBundler.js:74:12)
`

### Packages

- [X] create-react-native-library
- [X] react-native-builder-bob

### Selected options

Native module with Java and Obj-C

### Link to repro

https://github.com/Splicer97/react-native-mytracker

### Environment

System:
    OS: macOS 13.0
    CPU: (8) x64 Intel(R) Core(TM) i5-1038NG7 CPU @ 2.00GHz
    Memory: 35.35 MB / 16.00 GB
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.5.5 - /usr/local/bin/npm
    Watchman: 2022.09.05.00 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 16.0, macOS 12.3, tvOS 16.0, watchOS 9.0
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.3 AI-213.7172.25.2113.9123335
    Xcode: 14.0.1/14A400 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.16.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.5 => 0.70.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

    "react-native-builder-bob": "^0.20.0",

**Comment by MateWW** [comment]
The issue in your app is that you have changed the package name inside `package.json`

### Solution
- go to `tsconfig.json` in root and change `react-native-mytracker` with `@splicer97/react-native-mytracker`
- go to `App.tsx` swap `react-native-mytracker` with `@splicer97/react-native-mytracker`

### TL;DR;
We are using `name` field to link your package inside the example app however typescript needs a hardcoded name as it's json file.
As long as `tsconfig` has to be static `.json` we are limited to changing the name manually in both places.
